pep-0712.rst

```diff
@@ -0,0 +1,326 @@
+======================
+
+These changes don't introduce any compatibility problems since they
+only introduce opt-in new features.
+
+Security Implications
+======================
+
+There are no direct security concerns with these changes.
+
+How to Teach This
+=================
+
+Documentation and examples explaining the new parameter and behavior will be
+added to the relevant sections of the docs site (primarily on
+:mod:`dataclasses`) and linked from the *What's New* document.
+
+The added documentation/examples will also cover the "common pitfalls" that
+users of converters are likely to encounter. Such pitfalls include:
+* Needing to handle ``None``/sentinel values.
+* Needing to handle values that are already of the correct type.
+
+Reference Implementation
+========================
+
+The attrs library `already includes <attrs-converters_>`__ a ``converter``
+parameter containing converter semantics.
+
+CPython support is implemented on `a branch in the author's fork <cpython-branch_>`__.
+
+Rejected Ideas
+==============
+
+Just adding "converter" to ``typing.dataclass_transform``'s ``field_specifiers``
+--------------------------------------------------------------------------------
+
+The idea of isolating this addition to
+:func:`~py3.11:typing.dataclass_transform` was briefly
+`discussed on Typing-SIG <only-dataclass-transform_>`__ where it was suggested
+to broaden this to :mod:`dataclasses` more generally.
+
+Additionally, adding this to :mod:`dataclasses` ensures anyone can reap the
+benefits without requiring additional libraries.
+
+Not converting default values
+-----------------------------
+
+There are pros and cons with both converting and not converting default values.
+Leaving default values as-is allows type-checkers and dataclass authors to
+expect that the type of the default matches the type of the field. However,
+converting default values has two large advantages:
+
+1. Compatibility with attrs. Attrs unconditionally uses the converter to
+   convert the default value.
+
+2. Simpler defaults. Allowing the default value to have the same type as
+   user-provided values means dataclass authors get the same conveniences as
+   their callers.
+
+Automatic conversion using the field's type
+-------------------------------------------
+
+One idea could be to allow the type of the field specified (e.g. ``str`` or
+``int``) to be used as a converter for each argument provided.
+`Pydantic's data conversion <pydantic-data-conversion_>`__ has semantics which
+appear to be similar to this approach.
+
+This works well for fairly simple types, but leads to ambiguity in expected
+behavior for complex types such as generics. E.g. For ``tuple[int, ...]`` it is
+ambiguous if the converter is supposed to simply convert an iterable to a tuple,
+or if it is additionally supposed to convert each element type to ``int``.
+
+References
+==========
+.. _attrs-converters: https://www.attrs.org/en/21.2.0/examples.html#conversion
```

Comment:
```suggestion
==========

.. _attrs-converters: https://www.attrs.org/en/21.2.0/examples.html#conversion
```

Small formatting nit—couldn't make it last time since I was viewing the changes in "last-review-delta mode"